packages/builder/plugin-swc/src/plugin.ts

```diff
@@ -246,33 +303,47 @@ export class SwcWebpackPlugin {
 
         if (!minifiedSource) {
           const { source, map } = asset.source.sourceAndMap();
-          const minifyResult = await minifyWithTimeout(
-            asset.name,
-            source.toString(),
-            this.minifyOptions,
-          );
-
-          minifiedSource =
-            this.minifyOptions.sourceMap && minifyResult.map
-              ? new SourceMapSource(
-                  minifyResult.code,
-                  asset.name,
-                  minifyResult.map,
-                  source.toString(),
-                  map,
-                  true,
-                )
-              : new RawSource(minifyResult?.code || '');
+          let minifyResult: Output | undefined;
+          let needSourceMap = false;
+          const filename = asset.name;
+
+          if (filename.endsWith('.js') && jsMinify) {
```

Comment:
Reverse the condition order may be more performant.

```suggestion
          if (jsMinify && filename.endsWith('.js')) {
```